File: amplify/myPythonFunction/handler.js
const express = require("express");
const cors = require("cors");
const serverlessExpress = require("@vendia/serverless-express");

const app = express();

// CORS aktivieren (wichtig für die Kommunikation mit Nuxt)
app.use(
  cors({
    origin: "https://main.d3sa6211036fsg.amplifyapp.com", // URL des Nuxt-Projekts
    credentials: true,
  })
);

// API-Endpunkt
app.get("/", (req, res) => {
  res.json({ message: "Hello from Express!" });
});

// Exportieren des Lambda-Handlers
exports.handler = serverlessExpress({ app });

File: amplify/myPythonFunction/handler.py
from fastapi import FastAPI
from fastapi.middleware.cors import CORSMiddleware
from mangum import Mangum

app = FastAPI()

# CORS aktivieren (wichtig für die Kommunikation mit Nuxt)
app.add_middleware(
    CORSMiddleware,
    allow_origins=["https://main.d3sa6211036fsg.amplifyapp.com"],  # URL des Nuxt-Projekts
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

@app.get("/")
def read_root():
    return {"message": "Hello from FastAPI!"}

handler = Mangum(app)
